File: java/src/test/java/io/binac/leetcode/MinStackTest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class MinStackTest {

    @Test
    void test1() {
        MinStack minStack = new MinStack();
        minStack.push(-2);
        minStack.push(0);
        minStack.push(-3);
        assertEquals(-3, minStack.getMin());
        minStack.pop();
        assertEquals(0, minStack.top());
        assertEquals(-2, minStack.getMin());
    }
}

File: cpp/unittests/MinStackTest.cpp
#include "MinStack.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(MinStack, Solution155_1) {
  MinStack MStack;
  MStack.push(-2);
  MStack.push(0);
  MStack.push(-3);
  EXPECT_EQ(-3, MStack.getMin());
  MStack.pop();
  EXPECT_EQ(0, MStack.top());
  EXPECT_EQ(-2, MStack.getMin());
}